transformer_engine/common/util/cast_kernels.cuh

```diff
@@ -1194,12 +1196,21 @@ namespace detail {
 
 template <bool IS_DBIAS, bool IS_DACT, bool IS_ACT, typename ParamOP,
           float (*OP)(float, const ParamOP &)>
-void quantize_helper(const NVTETensor input, const NVTETensor activation_input,
-                     const NVTETensor noop, NVTETensor output, NVTETensor dbias,
-                     NVTETensor workspace, cudaStream_t stream) {
-  const auto &input_tensor = *(reinterpret_cast<const Tensor *>(input));
+void quantize_helper(const NVTETensor input, const NVTETensor grad, const NVTETensor noop,
+                     NVTETensor output, NVTETensor dbias, NVTETensor workspace,
+                     cudaStream_t stream) {
```

Comment:
The confusion from this function is not worth the code reuse. Better to split it up into three functions: `quantize_helper`, `forward_activation_helper`, `backward_activation_helper`.

Reply:
:100: